pkg/plugin/keyvault.go

```diff
@@ -94,37 +115,54 @@ func newKeyVaultClient(
 
 	klog.InfoS("using kms key for encrypt/decrypt", "vaultURL", *vaultURL, "keyName", keyName, "keyVersion", keyVersion)
 
-	client := &keyVaultClient{
+	client := &KeyVaultClient{
 		baseClient:       kvClient,
 		config:           config,
 		vaultName:        vaultName,
 		keyName:          keyName,
 		keyVersion:       keyVersion,
 		vaultURL:         *vaultURL,
 		azureEnvironment: env,
+		keyID:            fmt.Sprintf("%x", sha256.Sum256([]byte(fmt.Sprintf("%skeys/%s/%s", *vaultURL, keyName, keyVersion)))),
```

Comment:
Do we have a check somewhere that makes sure that `vaultURL` ends with a `/` ?

Reply:
Looking at the AKS docs the `kid` is a field on the actual key - would it be better to retrieve it instead of building it here?

```
export KEY_ID=$(az keyvault key show --name MyKeyName --vault-name MyKeyVault --query 'key.kid' -o tsv)
echo $KEY_ID
```

Reply:
This would lead to asking for more permission (List permission in addition to Encypt, and Decrypt) than we ask for. Keeping this as is to avoid it. Also, I have added a check in the encrypt method to make sure the keyID we are constructing matches the one we receive from encryptResponse.